src/plugins/userIndicator/components/UserIndicator.vue

```diff
@@ -65,18 +65,28 @@ export default {
       const UserAPI = this.openmct.user;
       const activeRole = UserAPI.getActiveRole();
       this.role = activeRole;
-      if (!activeRole) {
-        this.promptForRoleSelection();
+      this.availableRoles = await this.openmct.user.getPossibleRoles();
+
+      // clear role if it's not in list of available roles, e.g., removed by admin
+      if (!this.availableRoles?.includes(this.role)) {
+        this.role = null;
+        UserAPI.setActiveRole(null);
+      }
+
+      // see if we need to prompt for role selection
+      if (!this.role) {
+        this.promptForRoleSelection(this.availableRoles);
       } else {
         // only notify the user if they have more than one role available
-        this.availableRoles = await this.openmct.user.getPossibleRoles();
         if (this.availableRoles.length > 1) {
           this.openmct.notifications.info(`You're logged in as role ${activeRole}`);
         }
       }
     },
-    async promptForRoleSelection() {
-      this.availableRoles = await this.openmct.user.getPossibleRoles();
+    async promptForRoleSelection(passedAvailableRoles) {
```

Comment:
I'm not sure I see the advantage of the passedAvailableRoles since we always seem to have updated `this.availableRoles` before it is needed for the `selectionOptions` in the prompt.